app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,55 @@
+class VotesController < ApplicationController
+
+  skip_before_action :require_login
+
+  def create
+    work = Work.find_by(id: params[:work_id])
+    user = User.find_by(id: session[:user_id])
+    print "#{session[:user_id]}"
+    if work.nil?
+      flash.now[:error] = "Work not found"
+      head :not_found
+      return
+    elsif user.nil?
+      flash.now[:error] = "User not found"
+      head :not_found
```

Comment:
```suggestion
      flash[:error] = "You must be logged in to vote"
      redirect_back fallback_location: root_path
```
I honestly wish we'd never taught the use of `head` because we've since learned much better ways to handle these sorts of situations that I'd recommend you use instead (namely, `render` or `redirect_to`). I don't know if you ever tried this out, but it results in a generic browser error page so the user never sees the flash message.
In this specific case, because you'd ideally like to return to whatever page the user was already on, there's a nice method called `redirect_back` that a lot of students did a little internet searching to find :)